### PR TITLE
updating keras_for_text_classification.ipynb to keras3

### PR DESCRIPTION
**Pull request by sanjanalreddy**
- updated libraries to import from keras
- converted input to tensors since keras3 does not support https://github.com/keras-team/keras/issues/20479 imports anymore

### CODE REVIEW

File: notebooks/text_models/solutions/keras_for_text_classification.ipynb [hunk 1]
```diff
@@ -572,7 +594,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "X_train, X_valid = titles_train, titles_valid\n",
+    "X_train = tf.convert_to_tensor(X_train)\n",
+    "X_valid = tf.convert_to_tensor(X_valid)\n",
```
[the remark this answers is not in the document]
Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb]: Updated

File: notebooks/text_models/solutions/keras_for_text_classification.ipynb [hunk 2]
```diff
@@ -29,23 +29,25 @@
   {
```
Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>It seems this error happens when calling model.fit with GRU layer on GPU machine.</p><p> </p><p> </p><p><code>assertion failed: [You are passing a RNN mask that does not correspond to right-padded sequences, while using cuDNN, which is not supported. With cuDNN, RNN masks can only be used for right-padding, e.g. </code>[[True, True, False, False]]<code> would be a valid mask, but any mask that isn\'t just contiguous </code>True<code>\'s on the left and contiguous </code>False<code>\'s on the right would be invalid. You can pass </code>use_cudnn=False<code> to your RNN layer to stop using cuDNN (this may be slower).]</code></p><p><code>	 [[{{node sequential_1_1/gru_1/Assert/Assert}}]] [Op:__inference_multi_step_on_iterator_31700]</code></p><p> </p><p> </p><p>Adding <code>use_cudnn=False</code> to GRU layer resolves the issue, but there may be better way to handle it by modifying padding?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305220303'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='65'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>There's an equivalent ops in Keras.</p><p><code>keras.ops.convert_to_tensor</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305241800'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='39'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>There's an equivalent ops in Keras.</p><p><code>keras.ops.convert_to_tensor</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305241807'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='49'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #4.                Input(shape=(1,), dtype=tf.string, name="input_layer"),
</pre><p> </p><p><code>dtype='string'</code> in Keras. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256454'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='55'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #9.                Lambda(lambda x: tf.reduce_mean(x, axis=1)),  # TODO 4
</pre><p> </p><p><code>keras.ops.mean</code> in Keras.</p><p>Or we can replace the Lambda with <code>GlobalAveragePooling1D()</code> layer which does the same thing. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256458'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #3.    tf.random.set_seed(33)
</pre><p>Maybe remove this?</p><p>Or <code>keras.utils.set_random_seed</code> .</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256462'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='57'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #4.                Input(shape=(1,), dtype=tf.string),
</pre><p> </p><p><code>dtype='string'</code> in Keras. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256465'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='63'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #15.            optimizer=tf.keras.optimizers.Adam(learning_rate=0.0001),
</pre><p> </p><p>Please remove <code>tf.</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256470'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='63'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #4.                Input(shape=(1,), dtype=tf.string),
</pre><p> </p><p><code>dtype='string'</code> in Keras. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256474'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='69'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #21.            optimizer=tf.keras.optimizers.Adam(learning_rate=0.0001),
</pre><p> </p><p>Please remove <code>tf.</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256478'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='69'/>

Comment (takumiohym) [inline notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <pre class="ql-syntax" spellcheck="false">Line #3.    tf.random.set_seed(33)
</pre><p> </p><p>Maybe remove?</p><p>Or <code>keras.utils.set_random_seed</code> .</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/654/discussion/#comment-2305256482'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='71'/>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>I don't think this lab needs to run on GPU since it's only an introduction to text classification. So, I think modifying it to add use_cudnn=False is a good suggestion.</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>Thanks, updated! </p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>Updated</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>Done!</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>I will update to keras.ops.mean because it's a little more clear that it's a bag of words since GlobalAveragePool1D would use 1d windows which might lead to confusion with some folks as to why it's a DNN when we are using pooling layers.</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>Removed rather than update</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>done</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>done</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>Just a follow-up on modifying padding, I looked into it and it looks like <code>TextVectorization</code></p><p>handles right padding and so just the <code>embedding</code> layer, so not sure where there padding is getting messed up. For now, I have set cudnn to false </p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>done</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>done</p>

Comment (sanjanalreddy) [reply notebooks/text_models/solutions/keras_for_text_classification.ipynb:29]: <p>done</p>

**Review by takumiohym — APPROVED** [review]
LGTM! Thanks
